### ISSUE TITLE
Allow zoom on picture in mobile view

### ISSUE THREAD

**Issue by Globulle** [mhkeller/pulp2#10]
Sometimes it's not convenient to define panels on a page (one single fullpage picture for example). Then it becomes really hard to read on a mobile device. It could be interesting to allow page zooming. It could be useful to see details on a panel too.

**Comment by mhkeller** [comment]
This would be a little tricky but should be doable.

**Comment by mhkeller** [comment]
I kinda like how some sites do a pinch zoom and then when you release it, it comes back to normal size. Do you think that would work?

**Comment by mhkeller** [comment]
This one will be a bigger lift though since if you're already zooming in on mobile you have to stash those scale values.

**Comment by Globulle** [comment]
> Do you think that would work?

Yes I think that could really help reading the pages. Thanks!

**Comment by tsouk** [comment]
If I could upvote this issue, here it is: upvoted.

I think pinch zoom is the de facto way nowadays to look into more detail. Ideally we would have the option to do exactly what Instagram does, as people will default to that.

Also, the tap-swipe has confused almost every person I have tested this on.

**Comment by tsouk** [comment]
this looks promising, but I am having trouble adding it to the mix:
https://github.com/manuelstofer/pinchzoom

**Comment by mhkeller** [comment]
Ya natural behaviors have certainly matured since this first came out. What do you mean by tap-swipe? The panel-to-panel mobile navigation? What has your testing suggested as a more intuitive way?

**Comment by mhkeller** [comment]
@tsouk Lay out how you think the default should be since it sounds like the idea is to replace swiping from panel to panel is not coming across to readers. This issue was originally for just zooming in a little more but still preserving the panel-to-panel navigation.

How instagram currently works is you can zoom in with a two-finger pinch then if you release one finger you can pan. but you can't reach all of the image at that zoom level by panning. For this, are you imagining that from a page-view, a user two-finger pinches into their desired zoom level then releases a finger and pans from panel to panel? 

In that scenario, the amount the person moves their finger would have to be on a scale as opposed to one-to-one pixels. Let me know what you're thinking since an improvement would be good.

**Comment by tsouk** [comment]
Yes that is what I meant; that "tap & swipe" sentence was a bit mystical and it's because I have changed things, sorry.

But my point is that most people will try to pinch zoom on a mobile, even on an iPad, and that doesn't make the image bigger now, instead it zooms you to the first panel. Sometimes it happens by accident that a user will pinch-zoom on the first frame, the swipe event is picked up, and they are taken to the first frame believing that the pinch-zoom worked. So the next thing they do is try to pinch out. Then they are confused.

I love the current interaction but people actually need to be told/trained somehow, and I hate giving instructions over my comic book chapters.

I think the Instagram UX is also not ideal for comics. It doesn't allow for a relaxed reading when you zoom in as you need to have a finger on that picture all the time. The reason they've done it is so you can't easily screenshot things, I believe.

I think a good compromise would be similar to what Chrome mobile does when you load a plain picture:
- Pinch-zoom will zoom in and when you lift your fingers it keeps it zoomed. Then you can pan around with one finger.
- Double tap can zoom in a frame as single tap does now. We can use the nice masking features but it needs some UX testing. I think when zoomed in a frame if a user tries to pan the mask has to dissolve, so you can freely look around. Still this is the trickiest one.
- When zoomed in, double tap will always take you to the full page view.
- On full page view, swipe takes you forwards or backwards a page, as now.

Not 100% on this obviously, needs some guerilla testing.
I could ask my UX people at the BBC to give it a go, they might have some advice.

**Comment by tsouk** [comment]
Yes! Well broken down.
I would add to the last point

- [ ] Eliminate swipe panel-to-panel. When in pinch-zoom mode a swipe will pan, when not in pinch-zoom mode a swipe will take you to a different page

**Comment by mhkeller** [comment]
Sounds good. I started doing some of these on the [`new-ux`](https://github.com/mhkeller/pulp2/tree/new-ux) branch. On the double tap i get a screen flash that is a bit annoying...

**Comment by mhkeller** [comment]
fyi i added hammer.js to handle these mobile touch events

**Comment by ragtag** [comment]
First of, I love this tool. It's more or less just what I was looking for. I had actually started to roll my own version of it, when I came across yours.

I was testing the new-ux branch, and in it there seems to be no way to go between panels, other than going in and out, by double tapping. If you swipe, you go to the next page, which gets a bit confusing. I think comixology has solved this in a pretty nice way (they have some free comics, if you want to test it out).

The basic idea there is that you're either in what they call guided view, which is panel by panel, or full page view, where you can pinch to zoom.

- Double-tap on a panel in full page view, and it zooms in to that panel, and switches to guided view. Then swipe to advance panel by panel (optionally showing the full page first, before going into the first panel).
- Double-tap when in guided view, bring you back into full page view.
- In full page view you can pinch to zoom, and the use one finger to pan around the page. If you drag of the edge of the page, for half the screen width, it moves to the next page (zoomed out showing the whole page). If you double tap when zoomed in, you go back out to full page view.

It's a bit hard to explain, but it feels quite intuitive.

A bit of topic, but I also noticed that there is quite a bit of overlap in what Pulp2 does and the Advanced Comic Book Format ( http://acbf.wikia.com/wiki/Advanced_Comic_Book_Format_Wiki ). The latter is just for desktop, and the former for web. The ACBF xml files, contains more or less the information that Pulp2 would need to do it's job.
